oak/server/rust/oak_runtime/src/runtime/tests.rs

```diff
@@ -140,3 +140,68 @@ fn create_node_more_public_label() {
         }),
     );
 }
+
+#[test]
+fn wait_on_channels_immediately_returns_if_any_channel_is_orphaned() {
```

Comment:
Since the interesting part about the wait functionality is the waiting, would it be possible to have a test that does something concurrently? e.g. create two nodes, and close a channel after a while to make it orphaned, or send something on one of the channels while another node is waiting? I am not sure whether the test harness we currently have supports this, we may have to extend it to support running multiple nodes (maybe two nodes is enough?). Something for another PR perhaps, but could you create an issue and look into it @rbehjati ?

Reply:
I have added a new test that closes the write end of the channel in a separate thread. WDYT?

Reply:
Very neat! Thanks for doing this! Maybe we should think about a framework to make this kind of test even simpler in the future, e.g. running multiple node bodies in parallel. If you have any ideas, please create some GitHub issues to track.

Reply:
I agree. It would be more realistic to simulate the situation in this test with two parallel nodes. I'll think about it, especially if there are other scenarios that could benefit from such a utility.

Reply:
In "userspace" there's already [functionality for testing multi-Node applications](
https://github.com/project-oak/oak/blob/master/docs/programming-oak.md#testing-multi-node-applications); probably some commonization/re-use is possible.

Reply:
Nice. Thanks.